### PR TITLE
LPD-29744

### PR DESCRIPTION
**Pull request by rijema**
Please let me know if it's more appropriate to merge this test with the testGetObjectEntry. I'm introducing a new field and increasing the size of calls to follow the steps from the ticket.

### CODE REVIEW

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFieldResourceImpl.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public ObjectField postObjectDefinitionObjectField(
 			_objectDefinitionLocalService.getObjectDefinition(
 				objectDefinitionId);
 
-		return _toObjectField(
+		ObjectField objectField2 = _toObjectField(
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFieldResourceImpl.java:133]: Done.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 2]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
 		}
 	}
 
+	@Test
+	public void testGetUndeployedObjectFieldValue() throws Exception {
+		ObjectDefinition objectDefinition = _addObjectDefinition(true);
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:774]: Yes, it's a non-optional parameter of this private method :/

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 3]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
 		}
 	}
 
+	@Test
+	public void testGetUndeployedObjectFieldValue() throws Exception {
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:773]: Sure. Done.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 4]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
...
+		ObjectDefinition objectDefinition = _addObjectDefinition(true);
+
+		objectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				objectDefinition.getObjectDefinitionId());
+
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:780]: Done.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 5]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
...
+		String objectFieldName = StringUtil.randomId();
+
+		ObjectDefinition finalObjectDefinition = objectDefinition;
+
+		com.liferay.object.admin.rest.dto.v1_0.ObjectField objectField2 =
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:785]: I followed the approach of the number of fields. As _addObjectDefinition(true) already creates a field on the definition we will be using it, I added the suffix 2 to represent the quantity of fields, but yeah, I doesn't make sense if the code block itself doesn't contains any other object field.

[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:785]: Sure, done.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 6]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
...
+					).build();
+					localized = false;
+					name = objectFieldName;
+					objectDefinitionExternalReferenceCode1 =
+						finalObjectDefinition.getExternalReferenceCode();
+					required = false;
+					system = false;
+					unique = false;
+				}
+			};
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:808]: Unfortunately I won't be able to introduce all the random actions above as the localized version is necessary to remain false to this test to pass.  I talked in person about the _OBJECT_FIELD_NAME and why we won't be using it to name the new object field.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 7]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
...
+
+		ObjectFieldResource.Builder objectFieldResourceBuilder =
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:810]: Done

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 8]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
...
+
+		ObjectFieldResource objectFieldResource =
+			objectFieldResourceBuilder.user(
+				TestPropsValues.getUser()
+			).build();
+
+		objectFieldResource.postObjectDefinitionObjectField(
+			objectDefinition.getObjectDefinitionId(), objectField2);
+
+		ObjectEntry objectEntry = ObjectEntryLocalServiceUtil.addObjectEntry(
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:821]: Sure, done!

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java [hunk 9]
```diff
@@ -766,6 +769,86 @@ public void testGetObjectEntryWithLocalizedObjectField() throws Exception {
...
+							new GraphQLField(
+								key,
+								HashMapBuilder.<String, Object>put(
+									_getPKObjectFieldName(
+										finalObjectDefinition),
+									objectEntry.getObjectEntryId()
+								).build(),
+								new GraphQLField(objectFieldName))))),
+				"JSONObject/data", "JSONObject/c", "JSONObject/" + key,
+				"Object/" + objectFieldName));
```
[the remark this answers is not in the document]
Comment (rijema) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java:849]: Hi! Thanks for the insight! I updated the code properly with your requests and hopefully we can continue on this issue.

**Comment by rijema** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
@rijema please see https://github.com/liferay-objects/liferay-portal/pull/3552

**Pull request closed without merging**